zerver/lib/validator.py

```diff
@@ -189,7 +189,7 @@ def check_list(sub_validator: Optional[Validator], length: Optional[int]=None) -
 
     @set_type_structure(type_structure)
     def f(var_name: str, val: object) -> Optional[str]:
-        if not isinstance(val, list):
+        if not (isinstance(val, list) or isinstance(val, tuple)):
```

Comment:
What's the story with this change?  It seems unlikely to be correct in some sense; but I guess the issue is there's no difference between a tuple and a list in JSON?

Reply:
the schema check test mentioned in [this comment above](https://github.com/zulip/zulip/pull/13833#discussion_r375878827), was failiing.

Reply:
I would really want to avoid modifying `check_list` just to make it work for the somewhat unique muting use case.  We should get away from tuples on that API and just send a proper struct in.  Or, if we do keep tuples, just use a custom validator, so we can type-check the individual fields.  But we don't want any of the muting sloppiness to leak into other code.